src/utils/detect-server.js

```diff
@@ -176,7 +176,7 @@ const serverSettings = async (devConfig, flags, projectDir, log) => {
   settings.jwtRolePath = devConfig.jwtRolePath || 'app_metadata.authorization.roles'
   settings.functions = devConfig.functions || settings.functions
   if (settings.functions) {
-    settings.functionsPort = await getPort({ port: settings.functionsPort || 0 })
+    settings.functionsPort = await getPort({ port: devConfig.functionsPort || 0 })
```

Comment:
`settings.functionsPort` is always `undefined` since `settings` is returned from the framework detectors.
`devConfig.functionsPort` can be overridden in the `dev` block of `netlify.toml`